From 41bd9e8e46a97cfe8a0dfd2c2626aa7d4809ed4d Mon Sep 17 00:00:00 2001
From: LN Liberda <lauren@selfisekai.rocks>
Date: Wed, 28 May 2025 12:10:07 +0200
Subject: [PATCH] Allow to disable the Node.js version check with a build arg

For distributions building Chromium with their own Node.js binaries,
version mismatches are expected. Keep the check on by default,
but allow to opt-out of it.

Bug: 401522564
Change-Id: I211279a607b5b5fefd13614b177e55e57ba804c1
---
 third_party/node/BUILD.gn | 40 ++++++++++++++++++++++++---------------
 1 file changed, 25 insertions(+), 15 deletions(-)

diff --git a/third_party/node/BUILD.gn b/third_party/node/BUILD.gn
index 1741e14401c8c..75631fac8b462 100644
--- a/third_party/node/BUILD.gn
+++ b/third_party/node/BUILD.gn
@@ -4,22 +4,32 @@
 
 import("//third_party/node/node.gni")
 
-node("check_version") {
-  script = "check_version.py"
+declare_args() {
+  # Check the version of the Node.js binary.
+  node_version_check = true
+}
+
+if (node_version_check) {
+  node("check_version") {
+    script = "check_version.py"
 
-  expected_version_file = "update_node_binaries"
-  inputs = [
-    expected_version_file,
-    "check_version.js",
-  ]
+    expected_version_file = "update_node_binaries"
+    inputs = [
+      expected_version_file,
+      "check_version.js",
+    ]
 
-  out_file = "${target_gen_dir}/check_version_result.txt"
-  outputs = [ out_file ]
+    out_file = "${target_gen_dir}/check_version_result.txt"
+    outputs = [ out_file ]
 
-  args = [
-    "--expected_version_file",
-    rebase_path(expected_version_file, root_build_dir),
-    "--out_file",
-    rebase_path(out_file, root_build_dir),
-  ]
+    args = [
+      "--expected_version_file",
+      rebase_path(expected_version_file, root_build_dir),
+      "--out_file",
+      rebase_path(out_file, root_build_dir),
+    ]
+  }
+} else {
+  group("check_version") {
+  }
 }
